### ISSUE TITLE
Undo/Redo function in polygon-drawing

### ISSUE THREAD

**Issue by jo-spek** [google/ground-android#2699]
Hi, it is quite practical that there is an undo button in polygon mapping. 
However, if you accidentally undo one point too many, you can't "redo" it and it is lost. This can happen rather quickly, especially because of the lacking visual feedback after adding a corner (problem raised in issue https://github.com/google/ground-android/issues/2607). 
When mapping in the field, especially when walking from corner to corner with rather large distances, undoing a correct point can become really frustrating.

It would therefore be great to have an "undo" and "redo" button in the polygon mapping function. 

Visual suggestion:
<img src="https://github.com/user-attachments/assets/3afab862-46a5-4a5a-8a04-036e392ba855" width="270" height="600" />  -->  <img src="https://github.com/user-attachments/assets/1e0d7103-b975-457e-bb58-4571f4c78699" width="270" height="600" />

**Comment by kenstershiro** [comment]
Agreed with @anandwana001 and @shobhitagarwal1612 : Edge case we need to handle - if user adds another point before the next redo, where does the redo act? To simplify, suggestion is to clear the redo stack when the user adds another point/line.
Conserve the whole stack of the current polygon (one line at a time).
cc @jo-spek @gino-m

**Comment by gino-m** [comment]
Most "Undo" / "Redo" features clear the forward queue once a new thing is added. In other words, "redo" gets disabled as soon as a new point is added.
